src/edu/washington/escience/myria/operator/CSVFileScanFragment.java

```diff
@@ -179,23 +179,20 @@ public CSVFileScanFragment(
   @Override
   protected TupleBatch fetchNextReady() throws IOException, DbException {
     long lineNumberBegin = lineNumber;
-    boolean unfinishedEncapsulatedTokenRow = false;
+    boolean truncatedQuotedField = false;
 
     while ((buffer.numTuples() < TupleBatch.BATCH_SIZE) && !flagAsIncomplete) {
       lineNumber++;
       if (parser.isClosed()) {
         break;
       }
 
-      if (unfinishedEncapsulatedTokenRow) {
+      if (truncatedQuotedField) {
         onLastRow = true;
       }
 
-      try {
-        if (!onLastRow) {
-          record = iterator.next();
-        }
-      } catch (Exception e) {
+      if (!onLastRow) {
+        record = iterator.next();
```

Comment:
Oh I see. You are right. 

An even simpler example: If all workers read a very long row and worker number 1 just happens to have a truncated quoted first record. ```next()``` would be called first and it would break. 

I can test this and I think the solution might be to move the following statement inside the try block:

```
 if (!onLastRow) {
        record = iterator.next();
}
```